compileFile.js

```diff
@@ -2,34 +2,40 @@ const { spawn } = require("child_process");
 const preferences = require("./preferencesHelper");
 const path = require("path");
 const locationHelper = require("./locationHelper");
+const config = require("./config")
+
+/**
+ * Get flags which needed for compile based on language
+ * @param {string} language programming language
+ * @param {string} filePath complete path to .c, .cpp or .py file
+ */
+function getFlags(language, filePath) {
+    const ext = config.extensions[language];
+    let flags = preferences().get("compilationFlags" + ext).split(" ");
+    if (flags[0] === "")
+        flags = [];
+
+    if (language === 'Python')
+        flags = ['-m', 'compileall', '-b', filePath, ...flags];
```

Comment:
What is the need for the `-b` flag here? In-fact, with `python2`, it causes an error.

Reply:
The `-b` flag causes bytecode files to be written to their legacy location rather than `__pycache__` in `python3` but `python2` could work without it.

Reply:
Maybe the comment on the first answer [here](https://stackoverflow.com/questions/13511997/how-to-compile-all-py-file-to-pyc-file-in-a-folder-by-writing-a-python-script/28076884) will work.